### PR TITLE
Project proposal: EOF implementation (ETK language)

### PR DESCRIPTION
**Pull request by gzanitti**

### CODE REVIEW

File: projects/etk-eof-implementation.md
Comment (taxmeifyoucan) [inline projects/etk-eof-implementation.md:9]: "relevant EIPs..." since EOF is defined by multiple you linked in resources

File: projects/etk-eof-implementation.md
Comment (taxmeifyoucan) [inline projects/etk-eof-implementation.md:13]: Please add a sentence, maybe to motivation part, on why ETK implementation is needed. Maybe also shortly on what is ETK, you can add link to the toolkit repo

File: projects/etk-eof-implementation.md
Comment (taxmeifyoucan) [inline projects/etk-eof-implementation.md:61]: not even partial? @lightclient was working on some concept implementation I think and maybe JS client too. Not 100% sure about it but I think there was some work done on this before Shapella

**Review by taxmeifyoucan — COMMENTED** [review]
thanks for the project submission! Left some minor comments

**Review by taxmeifyoucan — APPROVED** [review]
Thanks for submitting the project! Just minor comments, I can go ahead and merge it if you are ready

File: projects/etk-eof-implementation.md
Comment (gzanitti) [reply projects/etk-eof-implementation.md:61]: There is some work here and there. For example, the Solidity repository has a branch with some progress, but it is not updated to the latest specification. My point was that, as the spec moves forward day by day, there is not yet a 100% finalized reference implementation. Anyway, maybe that was too strong a statement, I will rewrite it to make it clearer :)

**Comment by gzanitti** [comment]
> Thanks for submitting the project! Just minor comments, I can go ahead and merge it if you are ready

Thanks Mario. I made some modifications following your comments. Let me know if you consider that something else should be modified, otherwise it can be merged.

**Review by lightclient — APPROVED** [review]
Looks great. As a maintainer of ETK this would be really exciting! Excited to see your work.

**Comment by gzanitti** [comment]
> Looks great. As a maintainer of ETK this would be really exciting! Excited to see your work.


Hi @lightclient, thanks a lot for your feedback and enthusiasm! Since you're an ETK maintainer, could you also mentor me? Andrei would guide me on EOF specs, and you could help with ETK's codebase. Totally fine if you're busy, no pressure.

P.D: I came up with this idea after seeing that you were the one who approved my first pull request regarding MCOPY (EIP-5656)

**Comment by lightclient** [comment]
Yes I'm happy to help mentor you! I don't have as much time for etk these days but will try to help you navigate the code base as best I can. You can contact me on the eth r&d discord.

**Comment by gzanitti** [comment]
> Yes I'm happy to help mentor you! I don't have as much time for etk these days but will try to help you navigate the code base as best I can. You can contact me on the eth r&d discord.

That's awesome! Thank you so much! I'll include your name in my project proposal with another PR. I assure you that my interruptions will be minimal and only in strictly necessary situations :)

Thanks again!

**Pull request merged**
